README.md

```diff
@@ -456,3 +455,120 @@ ansible -i inventory.yaml overcloud -a "sudo dnf -y install python3 lvm2"
+
+In case a compute node gets removed, disable the Compute service on the outgoing node on the overcloud to prevent the node from scheduling new instances
+
+```bash
+openstack compute service list
+openstack compute service set <hostname> nova-compute --disable
+```
+
+## Annotate the VM/BMH resource for deletion
+
+Annotation of a VM resource
+
+```bash
+oc annotate -n openstack vm/controller-1 osp-director.openstack.org/delete-host=true --overwrite
+```
+
+Annotation of a BMH resource
+
+```bash
+oc annotate -n openshift-machine-api bmh/openshift-worker-3 osp-director.openstack.org/delete-host=true --overwrite
+```
+
+The annotation status is being reflected in the OSBaremetalset/OSVMset using the `annotatedForDeletion` parameter:
+
+```bash
+oc get osbms computehci -o json | jq .status
+{
+  "baremetalHosts": {
+    "computehci-0": {
+      "annotatedForDeletion": true,
+      "ctlplaneIP": "192.168.25.105/24",
+      "hostRef": "openshift-worker-3",
+      "hostname": "computehci-0",
+      "networkDataSecretName": "computehci-cloudinit-networkdata-openshift-worker-3",
+      "provisioningState": "provisioned",
+      "userDataSecretName": "computehci-cloudinit-userdata-openshift-worker-3"
+    },
+    "computehci-1": {
+      "annotatedForDeletion": false,
+      "ctlplaneIP": "192.168.25.106/24",
+      "hostRef": "openshift-worker-4",
+      "hostname": "computehci-1",
+      "networkDataSecretName": "computehci-cloudinit-networkdata-openshift-worker-4",
+      "provisioningState": "provisioned",
+      "userDataSecretName": "computehci-cloudinit-userdata-openshift-worker-4"
+    }
+  },
+  "provisioningStatus": {
+    "readyCount": 2,
+    "reason": "All requested BaremetalHosts have been provisioned",
+    "state": "provisioned"
+  }
+}
+```
+
+## Reduce the resource count
+
+Reducing the resource count of the OSBaremetalset will trigger the corrensponding controller to handle the resource deletion
+
+```bash
+oc patch osbms computehci --type=merge --patch '{"spec":{"count":1}}'
+```
+
+As a result:
+* the corresponding OSIPSet for the node got deleted
+* the IPreservation entry in the OSNet resources got flagged as deleted
+
+```bash
+oc get osnet ctlplane -o json | jq .status.roleReservations.ComputeHCI
+{
+  "addToPredictableIPs": true,
+  "reservations": [
+    {
+      "deleted": true,
+      "hostname": "computehci-0",
+      "ip": "192.168.25.105",
+      "vip": false
+    },
+    {
+      "deleted": false,
+      "hostname": "computehci-1",
+      "ip": "192.168.25.106",
+      "vip": false
+    }
+  ]
+}
+```
+
+This results in the following behavior
+* the IP is not free for use for another role
+* if a new node gets scaled into the same role it will reuse the hostnames starting with lowest id suffix (if there are multiple) _and_ corresponding IP reservation
+* if the OSBaremetalset or OSVMset resource gets deleted, all IP reservations for the role get deleted and are free to be used by other nodes
```

Comment:
Is this going to cause problems on the OOO side?  Just asking because my understanding was that we always needed to maintain past reservations, but maybe wiping a whole compute role or control plane role mitigates that concern?

Reply:
no, this does not cause issues. for existing roles we only need to make sure we have a persistent list of max predictable IPs list entries, like:
~~~
      ComputeHCIIPs:
        internal_api:
        - 172.16.2.9
        - 172.16.2.10
        - 172.16.2.11
        - 172.16.2.12
~~~

When compute-2 gets deleted we must keep an entry, not necessarily the IP reservation, to have:
~~~
      ComputeHCIIPs:
        internal_api:
        - 172.16.2.9
        - 172.16.2.10
        - deleted
        - 172.16.2.12
~~~